src/types.tsx

```diff
@@ -28,7 +28,7 @@ export interface PopupProps {
   nested?: boolean;
   defaultOpen?: boolean;
   on?: EventType | EventType[];
-  children: React.ReactNode;
+  children: React.ReactNode | ((close: () => void, isClose?: boolean) => React.ReactNode);
```

Comment:
```suggestion
  children: React.ReactNode | ((close: () => void, isOpen: boolean) => React.ReactNode);
```
Should it be called `isOpen` (instead of `isClose`)? And as you can see in line 306, the second param is always provided and cannot be undefined, so the `?` is unnecessary.

But otherwise the changes look good and should get merged.